Classes/DataProcessing/DatabaseQueryProcessor.php

```diff
@@ -126,22 +126,9 @@ public function process(ContentObjectRenderer $cObj, array $contentObjectConfigu
                     $processedRecordVariables[$key][$fieldName] = $overrideData;
                 }
             }
-
-            if ((int)($processorConfiguration['returnFlattenObject'] ?? 0)) {
-                $flattenRow = array_shift($processedRecordVariables);
-                break;
-            }
         }
 
-        if ($flattenRow !== null && (int)($processorConfiguration['returnFlattenLegacy'] ?? 1)) {
-            return $flattenRow;
-        }
-
-        if ($processorConfiguration['returnFlattenObject'] ?? 0) {
-            $processedData[$targetVariableName] = (int)($processorConfiguration['returnFlattenLegacy'] ?? 1) ? $processedRecordVariables : $flattenRow;
-        } else {
-            $processedData[$targetVariableName] = $processedRecordVariables;
-        }
+        $processedData[$targetVariableName] = $processorConfiguration['returnFlattenObject'] ?? 0 ? array_shift($processedRecordVariables) : $processedRecordVariables;
```

Comment:
that's a bit hard to read IMO. Why not do a proper if/then/else block? ;-)

Reply:
TBH is same for me, but if/else is more clear, I changed it